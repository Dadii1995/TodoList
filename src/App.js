import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import ProtectedRoute from './components/ProtectedRoute'
import Navigation from './components/Navigation'
import Profile from './containers/Profile'
import { Todos } from './containers/Todos'
import Blog from './containers/Blog'
import Weather from './containers/Weather'
import Contact from './containers/Contact'
import LoginForm from './containers/Auth/LoginForm'
import ThemeProvider from './contexts/Theme'
import AuthProvider from './contexts/Auth'

class App extends React.Component {

  render() {
    return (
      <AuthProvider>
        <ThemeProvider>
          <div className="App">
            <Router>
              <Navigation />
              <Switch>
                <ProtectedRoute component={Profile} path={['/user', '/profile']} />
                <ProtectedRoute component={Blog} path={'/blog'} />
                <ProtectedRoute component={Weather} path={'/weather'} />
                <ProtectedRoute component={Contact} path={'/contact'} />
                <Route component={LoginForm} path="/login" />
                <ProtectedRoute component={Todos} path={'/'} />
              </Switch>
            </Router>
          </div>
        </ThemeProvider>
      </AuthProvider>
    )
  }
}

export default App
