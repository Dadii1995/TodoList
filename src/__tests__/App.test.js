import renderer from 'react-test-renderer'
import React from 'react'
import App from '../App'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

jest.mock('../components/Navigation', () => () => <div data-testId="Navigation">Navigation</div>)
jest.mock('../containers/Profile', () => () => <div data-testId="Profile">Profile</div>)
jest.mock('../containers/Todos', () => () => <div data-testId="Todos">Todos</div>)
jest.mock('../containers/Blog', () => () => <div data-testId="Blog">Blog</div>)
jest.mock('../containers/Contact', () => () => <div data-testId="Contact">Contact</div>)
jest.mock('../containers/Weather', () => () => <div data-testId="Weather">Weather</div>)

const getWrapper = () => {
  const mockStore = configureStore()
  const store = mockStore()
  const wrapper = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  )
  const root = wrapper.root
  return { root, wrapper }
}
describe('<App/>', () => {
  test('Render App', () => {
    const { wrapper } = getWrapper()
    expect(wrapper).toMatchSnapshot()
  })
})
