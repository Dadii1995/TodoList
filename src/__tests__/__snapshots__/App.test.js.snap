// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Render App 1`] = `
<div
  className="App"
>
  <div
    data-testId="Navigation"
  >
    Navigation
  </div>
</div>
`;
